### ISSUE TITLE
C# intellisense does not detect unused variable if assigned from function call

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#27214]
If I assign a value to a variable, but never use that variable, VS will report:

	The variable 'x' is assigned, but its value is never used. 

For example:

```csharp
var one = 2;
```

This will report one as being a variable that was assigned but its value is never used.

If I assign a return value of a method into this variable, it will not report the same error. 

For example:

```csharp
var two = giveMeAValue();
```

If I never touch 'two' after this line, it will still not report the same error. 

I think this is a mistake.

If you do not use the value of the variable, regardless of where that value came from, it should be flagged as an error.

Here is a complete example:

```csharp
class Foobar {
	int giveMeAValue () {
		return 2;
	}

	public int test ( int c, int d ) {
		var one = 2;
		var two = giveMeAValue();

		return d + 1;
	}
}
```

The variable 'one' gets flagged. The variable 'two' is not flagged. (And as a sidenote, the function argument 'c' is not flagged either as being unused).

_This issue has been moved from https://developercommunity.visualstudio.com/content/problem/253209/c-intellisense-does-not-detect-unused-variable-if.html
VSTS ticketId: 616340_
_These are the original issue comments:_
(no comments)
_These are the original issue solutions:_
(no solutions)

**Comment by CyrusNajmabadi** [comment]
This is by design.  There are many who want and expect to be able to place the result of a computation somewhere so they can inspect it, even if it isn't something that is used later.  

It's possible this could become a warning in the future though if warning-waves come online.  Otherwise, it would be far too problematic a breaking change to warn about this.
